sql/rowexec/join_iters.go

```diff
@@ -841,9 +828,9 @@ func (i *lateralJoinIterator) Next(ctx *sql.Context) (sql.Row, error) {
 		row := i.buildRow(i.lRow, i.rRow)
 		i.rRow = nil
 		if i.cond != nil {
-			if res, err := i.cond.Eval(ctx, row); err != nil {
+			if res, err := sql.EvaluateCondition(ctx, i.cond, row); err != nil {
 				return nil, err
-			} else if res == false {
+			} else if sql.IsFalse(res) {
```

Comment:
Not sure whether this should be `!sql.IsTrue` like the other cases, but I've left it as IsFalse to keep to current behaviour

Reply:
This should definitely be `!sql.IsTrue` so that it handles null cases correctly.

If you fix that then this PR looks perfect.